extensions/amp-app-banner/1.0/amp-app-banner.js

```diff
@@ -0,0 +1,26 @@
+import {isExperimentOn} from '#experiments';
+
+import {userAssert} from '#utils/log';
+
+import {BaseElement} from './base-element';
+
+import {CSS} from '../../../build/amp-app-banner-1.0.css';
+
+/** @const {string} */
+const TAG = 'amp-app-banner';
+
+class AmpAppBanner extends BaseElement {
```

Comment:
Now here's the fun part: We need a way so that the Preact Component will use the AMP service when we're running the AMP mode, and the Bento services when running in Bento mode. The easiest way to accomplish this is to use a `AMPContext` and Preact's `Context` APIs to propagate the service downwards, so that we can change it on the fly to AMP services.

Reply:
Is there any precedent or examples of this architecture?  

From what I understand, Bento components should be decoupled from all AMP Services, right?

I have seen examples where the Bento component has certain overridable props, so that AMP-specific logic can be added (eg. templateService logic), but I have never seen a full AMP service passed/Context'd into a component, nor have I seen a Bento service that replaces an AMP service via Context.  It sounds like this is a larger architecture discussion.

Reply:
For reference, this discussion has moved to: https://github.com/ampproject/bento/discussions/5